driver.go

```diff
@@ -632,35 +618,85 @@ func memoryInBytes(strmem string) (int64, error) {
+	imageName := image
+	// If it is a shortname, we should not have to worry
+	// Let podman deal with it according to user configuration
+	if !shortnames.IsShortName(image) {
+		imageRef, err := parseImage(image)
+		if err != nil {
+			return imageID, fmt.Errorf("invalid image reference %s: %w\n", image, err)
 		}
+		// If its non-docker transport, we cannot ask for a pull or
+		// check for existence in the API without image plumbing
+		if imageRef.Transport().Name() != "docker" {
+			if imageRef.Transport().Name() != "oci-archive" &&
+				imageRef.Transport().Name() != "docker-archive" {
+				return imageID, fmt.Errorf("unssupported transport: %s", imageRef.Transport().Name())
+			}
+			// docker-archive:path[:docker-reference]
+			// oci-archive:path:tag
+			path := strings.Split(image, ":")[1]
+			d.logger.Debug("Load image archive", "path", path)
+			imageID, err = d.podman.ImageLoad(d.ctx, path)
+			if err != nil {
+				return imageID, fmt.Errorf("error while loading image: %w", err)
+			}
+			return imageID, nil
+		}
+		imageName = imageRef.DockerReference().String()
 	}
 
-	named, err := dockerref.ParseNormalizedNamed(image)
+	imageID, err := d.podman.ImageInspectID(d.ctx, imageName)
 	if err != nil {
-		return "", err
+		d.logger.Warn("Unable to check for local image", "image", imageName, "err", err)
+		// do NOT fail this operation, instead try to pull the
 	}
-
-	var tag, digest string
-
-	tagged, ok := named.(dockerref.Tagged)
-	if ok {
-		tag = tagged.Tag()
-		return fmt.Sprintf("%s:%s", named.Name(), tag), nil
+	if imageID != "" {
+		d.logger.Info("Found imageID", imageID, "for image", imageName, "in local storage")
+		return imageID, nil
 	}
 
-	digested, ok := named.(dockerref.Digested)
-	if ok {
-		digest = digested.Digest().String()
-		return fmt.Sprintf("%s@%s", named.Name(), digest), nil
+	d.logger.Debug("Pull image", "image", imageName)
+	if imageID, err = d.podman.ImagePull(d.ctx, imageName); err != nil {
+		return imageID, fmt.Errorf("failed to start task, unable to pull image %s : %w", imageName, err)
 	}
+	d.logger.Debug("Pulled image ID", "imageID", imageID)
+	return imageID, nil
+}
 
-	// Image is neither diggested, nor tagged. Default to 'latest'
-	return fmt.Sprintf("%s:%s", named.Name(), "latest"), nil
+func parseImage(image string) (types.ImageReference, error) {
+	var transport, name string
+	parts := strings.SplitN(image, ":", 2)
+	// In case the transport is missing, assume docker://
+	if len(parts) == 1 {
+		transport = "docker"
+		name = "//" + image
+	} else {
+		transport = parts[0]
+		name = parts[1]
+	}
+
+	// 1: name
+	switch transport {
+	case "docker":
+		return docker.ParseReference(name)
+	case "oci-archive":
+		return ociArchive.ParseReference(name)
+	case "docker-archive":
+		return dockerArchive.ParseReference(name)
+	default:
+		// We could have both an unknown/malformed transport
+		// or an image:tag with default "docker://" transport omitted
+		ref, err := docker.ParseReference("//" + image)
```

Comment:
Isn't this handled above in the `len(parts) == 1` check? I think we can just error here instead.

Reply:
Actually, the tests failed due to this, so I reverted it. The problem that we have here is in case somebody sends an image like `docker.io/library/busybox:latest` expecting the docker transport to be used as default. As the `:` is the same character used to separated the transport to the image to the tag, we have no efficient way to identify that in `parts`. 

I've changed the error handling a bit to reflect this a bit better. Is it more understandable now?